### PR TITLE
fix reading eeglab events if there is only one event in file

### PR DESCRIPTION
**Pull request by mmagnuski**
Hey there,
I've just had a TypeError with the eeglab event reader with a file that contains only one event. In such case the list comprehension that gathers event types fails because `'mat_struct' object is not iterable`.
This small change fixes the problem.

This is my first PR to mne-python so please be merciful :) If I messed something up I'll be happy to clean up, add something etc.

And just BTW - thanks for your efforts with mne-python, really cool package!

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
I will review once I'm back at a computer. 
Do you have a test ready?

**Comment by mmagnuski** [comment]
I was thinking about a test - should I upload an eeglab file with, for example one timesample and one event? Or is it better to modify (for example - programmatically) one of the files you already have?

**Comment by Eric89GXL** [comment]
Better to modify programmatically to keep data download sizes small

File: mne/io/eeglab/eeglab.py [hunk 1]
```diff
@@ -525,8 +525,13 @@ def _read_eeglab_events(eeg, event_id=None, event_id_func='strip_to_integer'):
     if event_id is None:
         event_id = dict()
 
-    types = [event.type for event in eeg.event]
-    latencies = [event.latency for event in eeg.event]
+    try:
+        types = [event.type for event in eeg.event]
+        latencies = [event.latency for event in eeg.event]
+    except TypeError:
```
Comment (Eric89GXL) [inline mne/io/eeglab/eeglab.py]: Instead of `try/except` (which we generally try to avoid) can you instead check to see if it's dim > 0?

Comment (jona-sassenhagen) [inline mne/io/eeglab/eeglab.py]: Something like

`types = [event.type for event in eeg.event] if len(eeg.event) > 1 else [eeg.event.type]`

or is that 'too clever' ..?

**Comment by mmagnuski** [comment]
Ok, I'll figure out a way to test it, I just now have problem with nose (not my nose though ;)) and I seem not to be able to upload it well because a python thread is running (should finish in a few hours).

File: mne/io/eeglab/eeglab.py (hunk 1 above)
Comment (mmagnuski) [inline mne/io/eeglab/eeglab.py]: Is `isinstance(eeg.event, np.ndarray)` acceptable?

Comment (Eric89GXL) [inline mne/io/eeglab/eeglab.py]: What types can they be?

Comment (mmagnuski) [inline mne/io/eeglab/eeglab.py]: `len(eeg.event)` gives an error with one event unfortunatelly, I tried that before.
The types can be:
- `np.ndarray` when no events (then `.shape` is (0,) ) or if more than one event (then `.shape` is nonzero)
- `scipy.io.matlab.mio5_params.mat_struct` if only one event

Comment (agramfort) [inline mne/io/eeglab/eeglab.py]: what is the type of eeg.event in your corner case?

Comment (Eric89GXL) [inline mne/io/eeglab/eeglab.py]: Must be `scipy.io.matlab.mio5_params.mat_struct`. So you could do:
```
if isinstance(event, np.ndarray):
    event = [event.type for event in eeg.event]
    ...
else:
    event = [eeg.event.type]
    ...
```

Comment (mmagnuski) [inline mne/io/eeglab/eeglab.py]: Yes, that's what I was proposing with `isinstance` - it works well, I've tested it with 0, 1 and more events :) Ok, I'll push the change.

**Comment by agramfort** [comment]
LGTM

but it's even better with a test

**Comment by mmagnuski** [comment]
I have a test that I didn't test yet (I have temporary problems with nose and it seems I need to wait for a python thread that is still running to stop) - I'll push it here, so you can say whether it is ok.

**Comment by Eric89GXL** [comment]
LGTM, ready from your end @mmagnuski ?

**Comment by jasmainak** [comment]
LGTM as well. Tests are really nice @mmagnuski !

**Comment by mmagnuski** [comment]
Yes, ready to merge :rocket: 
Thanks for your help and guidance!

**Comment by Eric89GXL** [comment]
Thanks @mmagnuski

**Pull request merged**